Cogs/Queue.py

```diff
@@ -0,0 +1,291 @@
+            if isinstance(ctx.channel, nextcord.Thread):
+                self.queues[channel_type]["ThreadId"] = ctx.channel.id
+                self.queues[channel_type]["ChannelId"] = ctx.channel.parent.id
+            elif isinstance(ctx.channel, nextcord.TextChannel):
+                # remove ThreadId in case the command was previously executed in a thread
+                self.queues[channel_type].pop("ThreadId", None)
+                self.queues[channel_type]["ChannelId"] = ctx.channel.id
+            else:
+                await utility.dm_user(ctx.author, 'Please place the queue in a text channel or thread')
+                return
+
+            queue_message = await ctx.send(embed=embed)
+            self.queues[channel_type]["MessageId"] = queue_message.id
+            self.queues[channel_type]["Entries"] = []
+
+            with open(self.QueueLocation, "w") as f:
+                json.dump(self.queues, f)
+            await self.helper.finish_processing(ctx)
+            print("-= The InitQueue command was used successfully by " + str(ctx.author.name) + " at " + str(
+                strftime("%a, %d %b %Y %H:%M:%S ", gmtime()) + "=-"))
+        else:
+            await utility.deny_command(ctx, "InitQueue")
+            await utility.dm_user(ctx.author, "This command is restricted to moderators")
+        await self.helper.log(f"{ctx.author.mention} has run the InitQueue command in {ctx.channel.mention}")
+
+    @commands.command()
+    async def Enqueue(self, ctx: commands.Context, channel_type: ChannelTypeParameter, script: str, availability: str,
+                      notes: typing.Optional[str]):
+        channel_type = utility.get_channel_type(channel_type)
+        users_in_queue = [entry["ST"]
+                          for entry in self.queues["Regular"]["Entries"] + self.queues["Experimental"]["Entries"]]
+        if ctx.author.id not in users_in_queue:
+            await utility.start_processing(ctx)
+            entry = {"ST": ctx.author.id, "Script": script, "Availability": availability}
+            if notes:
+                entry["Notes"] = notes
+            self.queues[channel_type]["Entries"].append(entry)
+            await self.update_queue_message(self.queues[channel_type])
+
+            with open(self.QueueLocation, "w") as f:
+                json.dump(self.queues, f)
+            await self.helper.finish_processing(ctx)
+            print("-= The Enqueue command was used successfully by " + str(ctx.author.name) + " at " + str(
+                strftime("%a, %d %b %Y %H:%M:%S ", gmtime()) + "=-"))
+        else:
+            await utility.deny_command(ctx, "Enqueue")
+            await utility.dm_user(ctx.author, "You may not join a text ST queue while you are already in one")
+
+        await self.helper.log(f"{ctx.author.mention} has run the Enqueue command")
+
+    @commands.command()
+    async def Dequeue(self, ctx: commands.Context):
```

Comment:
There are some small plans for a text & live queue to work from the same Bot, so I'd be more inclined to have the command be something along the "TextQueueLeave", not sold on that being the best name though, any ideas welcome